### PR TITLE
mpvScripts.mpris: 0.4 -> 0.5

### PR DESCRIPTION
**Pull request by doronbehar**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
@GrahamcOfBorg build mpvScripts.mpris

**Comment by doronbehar** [comment]
@GrahamcOfBorg build mpvScripts.mpris

**Comment by doronbehar** [comment]
@GrahamcOfBorg build mpvScripts.mpris

**Comment by doronbehar** [comment]
Last push is just a `git commit --amend` fixing the commit messages typos.

**Comment by doronbehar** [comment]
@GrahamcOfBorg build mpvScripts.mpris

**Comment by doronbehar** [comment]
Damn, I'm not at my best today. Sorry @GrahamcOfBorg but:

@GrahamcOfBorg build mpvScripts.mpris

**Comment by doronbehar** [comment]
@jtojnar this update includes the fix you helped me with at https://discourse.nixos.org/t/debug-why-a-derivation-has-a-reference-to-gcc/7009

File: pkgs/applications/video/mpv/scripts/mpris.nix
Comment (jtojnar) [inline pkgs/applications/video/mpv/scripts/mpris.nix:24]: I would just recommend installing it at `$out/share/mpv/scripts` (looking at the [arch package](https://www.archlinux.org/packages/community/x86_64/mpv/), that is where scripts are installed) and adding `share/mpv/scripts` to `stripDebugList` list. And also asking upstream to make the installation directory configurable.

Of course mpv expression might need to be customized.

Comment (doronbehar) [reply pkgs/applications/video/mpv/scripts/mpris.nix]: That's a good idea also considering automatic updates (because if `$out` is `$name` and `pname` is not set @r-ryantm doesn't pick it) and it's done in latest push. Regarding patching upstream's Makefile, that's an even better idea but should we wait then for their response for this update? Or perhaps write the patch and add it to our own derivation?

Comment (jtojnar) [reply pkgs/applications/video/mpv/scripts/mpris.nix]: No need to block this on their response, we can always cherry pick the commit once they merge it.

File: pkgs/applications/video/mpv/scripts/mpris.nix
Comment (jtojnar) [inline pkgs/applications/video/mpv/scripts/mpris.nix:25]: You can just use

```suggestion
    install -Dt $out/share/mpv/scripts mpris.so
```

Comment (doronbehar) [reply pkgs/applications/video/mpv/scripts/mpris.nix:25]: I used `install -Dt` upstream: https://github.com/hoyon/mpv-mpris/pull/38

File: pkgs/applications/video/mpv/scripts/mpris.nix
Comment (Mic92) [inline pkgs/applications/video/mpv/scripts/mpris.nix:16]:
```suggestion
      url = "https://github.com/hoyon/mpv-mpris/commit/5e00f275a666cc1aea1e93ae46f888293c6e502d.patch";
```

Comment (Mic92) [reply pkgs/applications/video/mpv/scripts/mpris.nix:16]: The pull request might change. The commit won't.

Comment (doronbehar) [reply pkgs/applications/video/mpv/scripts/mpris.nix]: :+1:. Didn't imagine the commit URL won't disappear ever.

Comment (doronbehar) [reply pkgs/applications/video/mpv/scripts/mpris.nix]: I mean, for e.g, the previous last commit of this PR is still alive: https://github.com/NixOS/nixpkgs/commit/595ad51476b3847a46eff4c638aa2635a8be1d8f.patch

Comment (Mic92) [reply pkgs/applications/video/mpv/scripts/mpris.nix]: github makes them immutable.

**Comment by doronbehar** [comment]
ping @Mic92 ?

File: pkgs/applications/video/mpv/scripts/mpris.nix
Comment (jtojnar) [inline pkgs/applications/video/mpv/scripts/mpris.nix:16]:
```suggestion
      url = "https://github.com/hoyon/mpv-mpris/commit/f1482350868bf20e4575f923943ec998469b255e.patch";
```

Comment (doronbehar) [reply pkgs/applications/video/mpv/scripts/mpris.nix]: :rocket:.

File: pkgs/applications/video/mpv/scripts/mpris.nix
Comment (jtojnar) [inline pkgs/applications/video/mpv/scripts/mpris.nix:30]: this is outdated

File: pkgs/applications/video/mpv/scripts/mpris.nix
Comment (jtojnar) [inline pkgs/applications/video/mpv/scripts/mpris.nix:13]: Move this closer to the patch

File: pkgs/applications/video/mpv/scripts/mpris.nix
Comment (jtojnar) [inline pkgs/applications/video/mpv/scripts/mpris.nix:23]: I see no mention of gobject-introspection, only glib.

Comment (doronbehar) [reply pkgs/applications/video/mpv/scripts/mpris.nix]: :rocket:

File: pkgs/applications/video/mpv/scripts/mpris.nix
Comment (doronbehar) [reply pkgs/applications/video/mpv/scripts/mpris.nix]: :guitar:

File: pkgs/applications/video/mpv/scripts/mpris.nix
Comment (doronbehar) [reply pkgs/applications/video/mpv/scripts/mpris.nix]: :umbrella:

**Review by jtojnar — APPROVED** [review]

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 86124` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
<br>- mpvScripts.convert
<br>- mpvScripts.mpris
</details>

**Pull request merged**
